### PR TITLE
Alternate worker culling strategy

### PR DESCRIPTION
**Pull request by ob-stripe**
### Description
This PR adds support for an alternate worker culling strategy: instead of culling the workers at the end of the internal worker list (which in all likelihood are the most recently started workers), setting the new `worker_culling_strategy` configuration option to `oldest` will cull the workers which were started the longest time ago.

This strategy makes it easier to implement a rolling restart of workers. This would be useful for us because our workers tend to leak memory over time. While of course the right thing to do would be write good code that does not leak memory :D in practice it's very hard and replacing workers every so often is a cheap and easy strategy.

### Your checklist for this pull request
- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [x] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [x] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [x] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: lib/puma/cluster.rb [hunk 1]
```diff
@@ -122,10 +129,10 @@ def cull_workers
...
     # @!attribute [r] next_worker_index
     def next_worker_index
-      all_positions =  0...@options[:workers]
-      occupied_positions = @workers.map { |w| w.index }
-      available_positions = all_positions.to_a - occupied_positions
-      available_positions.first
+      occupied_positions = @workers.map(&:index)
+      idx = 0
+      idx += 1 until !occupied_positions.include?(idx)
+      idx
```
Comment (ob-stripe) [inline lib/puma/cluster.rb:135]: The reason I changed this method's implementation is because I ran into an edge case while writing tests for this PR: if `@options[:workers]` is decreased (e.g. because the process received a TTOU signal), the previous implementation could potentially return `nil`.

**Review by MSP-Greg — APPROVED** [review]

File: lib/puma/dsl.rb [hunk 2]
```diff
@@ -795,6 +795,32 @@ def worker_shutdown_timeout(timeout)
       @options[:worker_shutdown_timeout] = Integer(timeout)
     end
 
+    # Set the strategy for worker culling.
+    #
+    # There are two possible values:
+    #
+    # 1. **:default** (the default) - the excess workers at the end of the
```
Comment (nateberkopec) [inline lib/puma/dsl.rb:802]: Let's call it `:youngest`. More descriptive.

**Comment by ob-stripe** [comment]
Renamed `default` to `youngest` as requested.
